kinoml/features/ligand.py

```diff
@@ -0,0 +1,179 @@
+    DICTIONARY : dict
+        Defines the character-integer mapping of the one-hot encoding.
+    """
+
+    DICTIONARY = {c: i for i, c in enumerate(
+        'BCFHIKNOPSUVWY'  # atoms
+        'acegilnosru'  # aromatic atoms
+        '-=#'  # bonds
+        '1234567890'  # ring closures
+        '.*'  # disconnections
+        '()'  # branches
+        '/+@:[]%\\'  # other characters
+        'LR$'  # single-char representation of Cl, Br, @@
+    )}
+
+    def __init__(self, molecule, pad_up_to=None, *args, **kwargs):
+        #super().__init__(*args, **kwargs)
+        self.molecule = molecule
+        self.pad_up_to = pad_up_to
+
+    def _featurize(self):
+        """
+        Featurizes ``self.molecule`` as a one-hot encoding of the SMILES representation.
+        If ``self.pad_up_to`` is defined, the padded version will be returned.
+#
+#        Returns
+#        =======
+#        np.array
+#            One-hot encoded SMILES, with shape (``len(self.DICTIONARY``, ``self.pad_up_to``, )).
+#
+#        Notes
+#        =======
+#        Double element symbols (such as Cl, Br for atoms and @@ for chirality) are replaced
+#        with single element symbols (L, R and $ respectively).
+#        """
+        ohe_matrix = np.zeros((len(self.DICTIONARY), len(self.molecule)))
+        for i, dict_char in enumerate(self.DICTIONARY):
+            for j, molecule_char in enumerate(self.molecule):
+                if dict_char == molecule_char:
+                    ohe_matrix[i,j] = 1
+        if self.pad_up_to is not None:
+            return np.pad(ohe_matrix, ((0,0), (0, self.pad_up_to-len(self.molecule))), mode='constant')
+        else: 
+            return ohe_matrix
+#        smiles = self.molecule.to_smiles().replace("Cl", "L").replace("Br", "R").replace("@@", "$")
+#        vec = np.array([self.DICTIONARY[c] for c in smiles])
+#        if self.pad_up_to is not None:
+#            vec = np.pad(vec, (0, self.pad_up_to - len(vec)))
+#        return tf.keras.utils.to_categorical(vec, len(self.DICTIONARY))
+
+
+class MorganFingerprintFeaturizer(_BaseFeaturizer):
+
+    """
+    Featurizes a ``Ligand`` using Morgan fingerprints bitvectors
+
+    Parameters
+    ==========
+    radius : int, optional=2
+        Morgan fingerprint neighborhood radius
+    nbits : int, optional=512
+        Length of the resulting bit vector
+    """
+
+    def __init__(self, molecule, radius=2, nbits=512, *args, **kwargs):
+        #super().__init__(*args, **kwargs)
+        self.radius = radius
+        self.nbits = nbits
+        self.molecule = molecule
+
+    def _featurize(self):
+        """
+        Featurizes ``self.molecule`` as a Morgan Fingerprint using RDKit
+
+        Returns
+        ========
+        np.array
+            Morgan fingerprint of radius 2 of molecule, with shape ``nbits``. 
+        """
+        from rdkit import Chem
+        from rdkit.Chem.AllChem import GetMorganFingerprintAsBitVect
+        #m = self.molecule.to_rdkit()
```

Comment:
We need to try to use those `Molecule` objects that are not yet defined. The `to_rdkit` logic should be encapsulated in those objects so the featurizer doesn't have to assume much (right now it's assuming that the ligand will always be a SMILES string). The `Ligand` objects are empty for now, but we can start filling them in as we find the need (and this is a good one!).